pkg/scalers/kafka_scaler.go

```diff
@@ -25,11 +25,12 @@ type kafkaScaler struct {
 }
 
 type kafkaMetadata struct {
-	bootstrapServers  []string
-	group             string
-	topic             string
-	lagThreshold      int64
-	offsetResetPolicy offsetResetPolicy
+	bootstrapServers       []string
+	group                  string
+	topic                  string
+	lagThreshold           int64
+	offsetResetPolicy      offsetResetPolicy
+	partitionLimitDisabled bool
```

Comment:
Generally ok, but I don't like the name of the property, isn't there a better option?

Reply:
@tomkerkhove do you have any opinion on the naming?

Reply:
I'm not a big fan of the naming either. Alternatives: `partitionCheckDisabled` or `allowIdlePartitions`

What about `allowIdlePartitions`? I got this from `Kafka, the definitive guide` which talks about idle partitions. It is also a common pattern to keep idle consumers for HA.

Reply:
Sounds better imho, it should come with a solid docs though.

Reply:
LGTM

Reply:
Ok great! I will update this PR and create a new one for the doc.
Thanks

Reply:
Oops sorry I meant to type `allowIdleConsumers`, which is more accurate. LMK if you object to the change.